### PR TITLE
Add about Rector documentation

### LINKED ISSUE #70: Essential informations are missing in the introduction
|    Q        |   A
|------------ | ------
| Version(s)  | 0.4.0

The introduction starts with:

> # laminas-servicemanager-migration
>
> ## Using with Rector
>
> To use with rector, you can create the following `rector.php`:

Some informations are missing in this introduction:

- [x] What is Rector?
- [ ] How can it help with laminas-servicemanager?
- [ ] Which options / rules are available?
- [x] Link to the official website or documentation

([Looks like](https://getrector.org) that the official notation is "Rector", not "rector".)

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| Documentation | yes

### Description

@froschdesign this is copied from the website https://getrector.org/about

Ref to fill part of https://github.com/laminas/laminas-servicemanager-migration/issues/70

### CODE REVIEW

File: docs/book/v1/intro.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # laminas-servicemanager-migration
 
+## About Rector
+
+Rector is a PHP tool that you can run on any PHP project to get instant upgrade or automated refactoring. It helps with PHP upgrade, framework upgrade and improves your code quality. Also it helps with type-coverage and help to get to the latest PHPStan level.
```
Comment (froschdesign) [inline docs/book/v1/intro.md]:
```suggestion
Rector is a console application for automated refactoring of PHP applications.
It helps with:

- PHP upgrades
- Framework upgrades
- Code quality improvements
- Type coverage

For more information about Rector, visit: [getrector.org](https://getrector.org)
```

**Comment by samsonasik** [comment]
@froschdesign updated 👍

File: docs/book/v1/intro.md [hunk 2]
```diff
@@ -1,8 +1,20 @@
 # laminas-servicemanager-migration
```
Comment (froschdesign) [inline docs/book/v1/intro.md]: Navigation entry and title / first headline of the page must be the same:

https://github.com/laminas/laminas-servicemanager-migration/blob/751890e8ac17cb9f806359945845abe6fa02b200/mkdocs.yml#L5

```suggestion
# Introduction
```

**Review by froschdesign — CHANGES_REQUESTED** [review]
An important information is missing in the introduction:

How can it help with laminas-servicemanager?

**Comment by samsonasik** [comment]
Yes, that still missing, should probably separate PR for that, per PR description, it only part of #70

**Comment by samsonasik** [comment]
Let's merge so we have progress

**Comment by samsonasik** [comment]
@froschdesign thank you for the review.

**Pull request merged**